documentation/src/docs/admin-guide/deploying-small-bookkeeper-clusters.md

```diff
@@ -0,0 +1,75 @@
+
+However, there may be cases in which users need to deploy Pravega
+(and Bookkeeper) in small clusters (i.e., very limited number of nodes
+and/or storage per node). Note that a "node" can be either a physical or
+a virtual entity running Bookies. In this scenario, the decisions taken to 
+deploy Bookkeeper and configure the Bookkeeper client in Pravega may have 
+important implications on, at least, the following key aspects:
+
+- _Tolerance to temporary failures_: Whether a temporary failure in a 
+node or a Bookie implies that Pravega can continue reading (_R_) or writing
+(_W_) data to Bookkeeper. For instance, if a given Bookkeeper cluster and 
+client configuration can tolerate one temporary Bookie failure while writing
+data, we could denote it as `1W/0R`. Similarly, if a client can read data
+while a Bookie is unavailable, we denote it as `0W/1R`. This aspect is
+relevant, as it determines whether Pravega can continue performing IO while
+one or more Bookies/nodes are not reachable for some time.
+
+- _Tolerance to permanent failures_: Apart from temporary unavailability of
+Bookies and nodes, users may also face permanent crashes that imply the
+loss of data of a given Bookie or even a whole node. This is a critical
+aspect to take into account, as in general we should not allow data to 
+be lost.
+
+- _Performance_: Bookkeeper stores data using two main storage locations:
+_journal_ and _ledger_ (we can leave _index_ aside in this discussion).
+The main reason for having two storage locations for data is the different
+types of workloads they exhibit (e.g., sequential, synchronous writes
+vs buffered, random writes plus reads). This design decision is key
+if using hard drives, as it allows us to use one dedicated drive per workload
+type to improve performance. But even using modern SSD or NVMe drives for Bookies,
+users need to decide whether to dedicate individual drives for journal and ledger
+or collocating them in the same drive. This still plays a relevant role in performance.
+
+In the table below, we have summarized the trade-offs that different deployment
+options of Bookkeeper and client configuration may have in small clusters.
+We highlight the following takeaways:
+- In many cases, you need to chose between tolerating temporary write failure  
+or reads failures. This is because for very small clusters, the replication factor
+configured in the client is the same as the number of Bookies, so no write failure
+is allowed (but we can still read data and tolerate permanent Bookie crashes).
+- Some cluster layouts force users to trade-off between good performance 
+(i.e., dedicating 1 drive to each journal and ledger) and tolerate permanent crashes.
+- [Topology-awareness in Bookkeeper client](https://bookkeeper.apache.org/docs/latest/api/javadoc/org/apache/bookkeeper/client/EnsemblePlacementPolicy.html)
+is very important to prevent data loss when we have very few nodes to run Bookies.
+
```

Comment:
You could cite the "Opportunistic striping" feature, that allows you to use bigger ensemble sizes in case you have all (most of) the bookies available.
This helps a lot in case of small clusters, say with 3 bookies, because with this feature you can still use ensembleSize=3, WQ=2 and AQ=2 and tolerate a temporary failure of one bookie (in that case the ledgers created during the temporary failure will be written using 2-2-2 instead of 3-2-2)

Reply:
Thanks, added this comment in "final remarks" section.